internal/api/registry/blobs.go

```diff
@@ -118,7 +118,7 @@ func (a *API) handleGetOrHeadBlob(w http.ResponseWriter, r *http.Request) {
 		//AnycastBlobBytePullAction is only relevant for GET requests since it
 		//limits the size of the response body (which is empty for HEAD)
 		if r.Method == http.MethodGet {
-			err = api.CheckRateLimit(r, a.rle, *account, authz, keppel.AnycastBlobBytePullAction, blob.SizeBytes)
+			err = api.CheckRateLimit(r, a.rle, *account, authz, keppel.AnycastBlobBytePullAction, int(blob.SizeBytes))
```

Comment:
Either way we hider here that the rate limiter caps at 2.147 GB or max int32 aka 2147483647 bytes.